packages/frontend/web/components/elements/ImageBlockComponent.tsx

```diff
@@ -58,24 +61,110 @@ const getFallback: (imageSources: ImageSource[]) => string = imageSources => {
     return bestFor(300, inlineSrcSets).src;
 };
 
+const decidePosition = (role: RoleType) => {
+    switch (role) {
+        case 'immersive':
+            return css``; // TODO pending review of different article types
+            break;
+        case 'supporting':
+            return css`
+                ${tablet} {
+                    position: relative;
+                    float: left;
+                    width: 300px;
+                    margin-top: 6px;
+                    margin-bottom: 12px;
+                    margin-right: 20px;
+                    line-height: 0;
+                }
+                ${leftCol} {
+                    margin-left: -160px;
+                }
+                ${wide} {
+                    width: 380px;
+                    margin-left: -240px;
+                }
+            `;
+            break;
+        case 'showcase':
+            // TODO: This is only a partial implementation as sometimes showcase images
+            //       appear differently based on where they appear in the article
+            //       See top image for: https://www.theguardian.com/stage/2019/oct/01/hammed-animashaun-nick-hytner-dream-master-harold-and-the-boys-national-theatre-london
+            return css`
+                position: relative;
+                margin-top: 16px;
+                margin-bottom: 12px;
+                ${leftCol} {
+                    position: relative;
+                    margin-bottom: 16px;
+                    margin-left: -160px;
+                }
+                ${wide} {
+                    margin-left: -240px;
+                }
+            `;
+            break;
+        case 'inline':
+            return css`
+                margin-top: 16px;
+                margin-bottom: 12px;
+            `;
+            break;
+        case 'thumbnail':
+            return css`
+                float: left;
```

Comment:
Is it worth lifting these css definitions out of the switch to make the switch itself more readable?

Reply:
It is worth doing this. Done.